pandas/core/index.py

```diff
@@ -1548,8 +1548,15 @@ def sym_diff(self, other, result_name=None):
         other, result_name_update = self._convert_can_do_setop(other)
         if result_name is None:
             result_name = result_name_update
-        the_diff = sorted(set((self.difference(other)).union(other.difference(self))))
-        return Index(the_diff, name=result_name)
+        the_diff = self.difference(other).union(other.difference(self))
+        the_diff_sorted = np.sort(the_diff.unique())
```

Comment:
why not ``self._shallow_copy(the_diff)`` (and use the existing ``the_diff`` from the original code).